

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Quality trimming Illumina paired-end reads &mdash; Metagenomics Workshop 0.1 documentation</title>
    
    <link rel="stylesheet" href="_static/default.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '',
        VERSION:     '0.1',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="Metagenomics Workshop 0.1 documentation" href="index.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li><a href="index.html">Metagenomics Workshop 0.1 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="quality-trimming-illumina-paired-end-reads">
<h1>Quality trimming Illumina paired-end reads<a class="headerlink" href="#quality-trimming-illumina-paired-end-reads" title="Permalink to this headline">¶</a></h1>
<div class="section" id="sickle">
<h2>Sickle<a class="headerlink" href="#sickle" title="Permalink to this headline">¶</a></h2>
<p>For quality trimming Illumina paired end reads we use the library sickle which
trims reads from 3&#8217; end to 5&#8217; end using a sliding window. If the mean quality
drops below a specified number remaining part of the read will be trimmed.</p>
</div>
<div class="section" id="sickle-on-uppmax">
<h2>Sickle on Uppmax<a class="headerlink" href="#sickle-on-uppmax" title="Permalink to this headline">¶</a></h2>
<p>Daniel has already installed sickle in the project dir on uppmax so hurray! The
absolute path is <tt class="docutils literal"><span class="pre">/proj/b2010008/bin/sickle</span></tt>. <tt class="docutils literal"><span class="pre">/proj/b2010008/bin`</span></tt> to your
<tt class="docutils literal"><span class="pre">PATH</span></tt> variable if you haven&#8217;t done so already:</p>
<div class="highlight-python"><pre>export PATH=$PATH:/proj/b2010008/bin</pre>
</div>
<p>Note that this sets the variable only for your current shell and all
subprocesses spawned by it. If you want the directory added to your path for
always add the command to the file <tt class="docutils literal"><span class="pre">~/.bashrc</span></tt> which is executed everytime bash
is started.</p>
</div>
<div class="section" id="optional-install-sickle-by-yourself">
<h2>(Optional) Install sickle by yourself<a class="headerlink" href="#optional-install-sickle-by-yourself" title="Permalink to this headline">¶</a></h2>
<p>Follow these steps only if you want to install sickle by yourself. We are not
doing this for the workshop today, but I included it just for future reference.
Installation procedures of research software are often follow the same pattern.
Download the code, <em>compile</em> it and copy the binary to a location in your
<tt class="docutils literal"><span class="pre">PATH</span></tt>. The code for sickle is on <a class="reference external" href="https://github.com/najoshi/sickle">https://github.com/najoshi/sickle</a>. I prefer
<em>compiling</em> my programs in <tt class="docutils literal"><span class="pre">~/src</span></tt> and then copying the resulting program to
my <tt class="docutils literal"><span class="pre">~/bin</span></tt> directory, which is in my <tt class="docutils literal"><span class="pre">PATH</span></tt>. This should get you a long
way:</p>
<div class="highlight-python"><pre>mkdir -p ~/src

# Go to the source directory and clone the sickle repository
cd ~/src
git clone https://github.com/najoshi/sickle
cd sickle

# Compile the program
make

# Create a bin directory
mkdir -p ~/bin
cp sickle ~/bin</pre>
</div>
</div>
<div class="section" id="downloading-a-test-set">
<h2>Downloading a test set<a class="headerlink" href="#downloading-a-test-set" title="Permalink to this headline">¶</a></h2>
<p>Today we&#8217;ll be working on a small metagenomic data set from the anterior nares
(<a class="reference external" href="http://en.wikipedia.org/wiki/Anterior_nares">http://en.wikipedia.org/wiki/Anterior_nares</a>).</p>
<img alt="_images/nostril.jpg" src="_images/nostril.jpg" />
<p>So get ready for your first smell of metagenomic assembly - O, yes I did. Run
all these commands in your shell:</p>
<div class="highlight-python"><pre># Download the reads and extract them
mkdir -p ~/glob/asm-workshop/data
cd ~/glob/asm-workshop/data
wget http://downloads.hmpdacc.org/data/Illumina/anterior_nares/SRS018585.tar.bz2
tar -xjf SRS018585.tar.bz2</pre>
</div>
<p>If successfull you should have the files
<tt class="docutils literal"><span class="pre">~/asm-workshop/data/SRS018585/SRS018585.denovo_duplicates_marked.trimmed.1.fastq</span></tt>
and
<tt class="docutils literal"><span class="pre">~/asm-workshop/data/SRS018585/SRS018585.denovo_duplicates_marked.trimmed.2.fastq</span></tt>.</p>
</div>
<div class="section" id="running-sickle-on-a-paired-end-library">
<h2>Running sickle on a paired end library<a class="headerlink" href="#running-sickle-on-a-paired-end-library" title="Permalink to this headline">¶</a></h2>
<p>I like to create directories for specific parts I&#8217;m working on and creating
symbolic links (shortcuts in windows) to the files so there we go:</p>
<div class="highlight-python"><pre>mkdir -p ~/glob/asm-workshop/sickle
cd ~/glob/asm-workshop/sickle
ln -s ../data/SRS018585/SRS018585.denovo_duplicates_marked.trimmed.1.fastq pair1.fastq
ln -s ../data/SRS018585/SRS018585.denovo_duplicates_marked.trimmed.1.fastq pair2.fastq</pre>
</div>
<p>The difference between a symbolic link a and a hard link can be found here:
<a class="reference external" href="http://stackoverflow.com/questions/185899/what-is-the-difference-between-a-symbolic-link-and-a-hard-link">http://stackoverflow.com/questions/185899/what-is-the-difference-between-a-symbolic-link-and-a-hard-link</a>.
In this case I use symbolic links so I know what path the original reads have,
which can help one remember what those reads were after an intoxicating night
out. Now run sickle:</p>
<div class="highlight-python"><pre># check if sickle is in your PATH
which sickle
# Run sickle
sickle pe \
    -f pair1.fastq \
    -r pair2.fastq \
    -t sanger \
    -o qtrim1.fastq \
    -p qtrim2.fastq \
    -s qtrim.unpaired.fastq
# Check what files have been generated
ls</pre>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Quality trimming Illumina paired-end reads</a><ul>
<li><a class="reference internal" href="#sickle">Sickle</a></li>
<li><a class="reference internal" href="#sickle-on-uppmax">Sickle on Uppmax</a></li>
<li><a class="reference internal" href="#optional-install-sickle-by-yourself">(Optional) Install sickle by yourself</a></li>
<li><a class="reference internal" href="#downloading-a-test-set">Downloading a test set</a></li>
<li><a class="reference internal" href="#running-sickle-on-a-paired-end-library">Running sickle on a paired end library</a></li>
</ul>
</li>
</ul>

  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="_sources/qtrim.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li><a href="index.html">Metagenomics Workshop 0.1 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2013, Ino de Bruijn, Marcela Davila, Intawat Nookaew.
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 1.1.3.
    </div>
  </body>
</html>